### PR TITLE
Milestone5(event handlers) completed.

### PR DESCRIPTION
**Pull request by noomdalv**
**React Calculator is Functional**
## Implementations >>

**App component:**

- [x] Added constructor with state(total, next, operation) properties set to null.
- [x] Added handleClick function and passed as prop to ButtonPanel.

**ButtonPanel component:** 

- [x] Added handleClick function and passed as prop to Button.

**Button component:**

- [x] Added onClick event handler with the handleClick function.

### CODE REVIEW

**Review by jofer86 — CHANGES_REQUESTED** [review]
## :wave: :wave: Hello!, I'm Jorge, and  I'm here to review this Pull Request, let's take a look shall we :eye: :eye: :eye: 
- Changes marked as **[OPTIONAL]** are not hard requirements but are nice to have.
- All of the requirements are present in your code, however, we are missing one key component into the calculator functionality, and that is handling the error when you try to divide by 0, right now, if I divide by zero the entire application crashes, please do see to tend to that and catch that error and display error on the screen, you need to implement some error handling there to prevent the crash, do that and this project should be approved.

--------------------------------------------------------------------
Cheers, 
Jorge Fernando Rincón Figueroa.
Technical Support Engineer.
**Slack: @jofer86**

**Comment by noomdalv** [comment]
Added catch block in operate.js division case for zero division error, should throw an alert now

**Comment by bellom** [comment]
> Added catch block in operate.js division case for zero division error, should throw an alert now

You did fix the issue but the best practice is to display "error" as the result.

Don't forget it is not encouraged to use `alert` in production.

File: src/logic/operate.js [hunk 1]
```diff
@@ -8,7 +8,12 @@ const operate = (numberOne, numberTwo, operation) => {
 
   switch (operation) {
     case ('÷'):
-      result = a.div(b).toFixed(2).toString();
+			try {
+				result = a.div(b).toFixed(2).toString();
+			} catch(e) {				
+				alert("Division by zero error");
```
Comment (bellom) [inline src/logic/operate.js:14]: You did fix the issue but the best practice is to display "error" as the result.

Don't forget it is not encouraged to use alert in production.

File: .eslintrc.json [hunk 2]
```diff
@@ -17,6 +17,7 @@
     "react/jsx-filename-extension": ["warn", { "extensions": [".js", ".jsx"] }],
     "import/no-unresolved": "off",
     "no-shadow": "off",
-    "arrow-parens": ["error", "as-needed"]
+    "arrow-parens": ["error", "as-needed"],
+		"react/forbid-prop-types": "off"
```
Comment (bellom) [inline .eslintrc.json:21]: **NOTE:**
**1. In the next project, make use of microverse linter config, you're not supposed to edit it.**
You made some changes in this file, check out the [microverse-linter-config](https://github.com/microverseinc/linters-config/blob/master/react-redux/.eslintrc.json)
image

**Review by bellom — APPROVED** [review]
## Hi @noomdalv  :clap: 

**Great work making all the necessary corrections from the previous review.**

## Congratulation :+1:  
_Go ahead and merge to dev and don't forget to submit project completion form.
Good luck in the next project :slightly_smiling_face:_ 

### Take note of the comments below when working in your next project

Best,

**Oluwaseun Bello**
**_Slack Handle - @sean bellom_**

**Pull request merged**